tamr_client/mastering/_mastering.py

```diff
@@ -10,80 +10,173 @@
+    operation to complete
 
     Args:
         project: Tamr Mastering project
     """
-    r = session.post(str(project.url) + "recordPairsWithPredictions/model:refresh")
-    return operation._from_response(project.url.instance, r)
+    op = _apply_feedback_async(session, project)
+    return operation.wait(session, op)
 
 
 def update_pair_results(session: Session, project: MasteringProject) -> Operation:
-    """Updates record pair predictions according to the latest pair-matching model
+    """Update record pair predictions according to the latest pair-matching model and
+    wait for the operation to complete
 
     Args:
         project: Tamr Mastering project
     """
-    r = session.post(str(project.url) + "recordPairsWithPredictions:refresh")
-    return operation._from_response(project.url.instance, r)
+    op = _update_pair_results_async(session, project)
+    return operation.wait(session, op)
 
 
 def update_high_impact_pairs(session: Session, project: MasteringProject) -> Operation:
-    """Produces new high-impact pairs according to the latest pair-matching model
+    """Produce new high-impact pairs according to the latest pair-matching model and
+    wait for the operation to complete
 
     Args:
         project: Tamr Mastering project
     """
-    r = session.post(str(project.url) + "highImpactPairs:refresh")
-    return operation._from_response(project.url.instance, r)
+    op = _update_high_impact_pairs_async(session, project)
+    return operation.wait(session, op)
 
 
 def update_cluster_results(session: Session, project: MasteringProject) -> Operation:
-    """Generates clusters based on the latest pair-matching model
+    """Generate clusters based on the latest pair-matching model and wait for the
+    operation to complete
 
     Args:
         project: Tamr Mastering project
     """
-    r = session.post(str(project.url) + "recordClusters:refresh")
-    return operation._from_response(project.url.instance, r)
+    op = _update_cluster_results_async(session, project)
+    return operation.wait(session, op)
 
 
 def publish_clusters(session: Session, project: MasteringProject) -> Operation:
-    """Publishes current record clusters
+    """Publish current record clusters and wait for the operation to complete
+
+    Args:
+        project: Tamr Mastering project
+    """
+    op = _publish_clusters_async(session, project)
+    return operation.wait(session, op)
+
+
+def _update_unified_dataset_async(
+    session: Session, project: MasteringProject
+) -> Operation:
+    """Apply changes to the unified dataset
+
+    Args:
+        project: Tamr Mastering project
+    """
+    unified_dataset = unified.from_project(session, project.url.instance, project)
+    return unified._apply_changes_async(session, unified_dataset)
+
+
+def _estimate_pairs_async(session: Session, project: MasteringProject) -> Operation:
+    """Update the estimated pair counts
+
+    Args:
+        project: Tamr Mastering project
+    """
```

Comment:
I think docstrings for these functions is overkill given that they only have one usage that is quite obvious and they are well-named.

I lean towards removing docstrings for these functions, but will leave the decision to you.